### PR TITLE
Riya/search slash menu/prod

### PR DESCRIPTION
**Pull request by ViaSocket-Git**

### CODE REVIEW

File: src/appPages/page/page.jsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ import { useRouter, useParams } from "next/navigation";
 import { navigateTo } from "src/navigationService";
 import { setPagesPath } from "../../components/pages/redux/pagesActions";
 import TagInput from "@/components/publishModal/tagInput";
-import { updatePage } from "@/components/pages/pageApiService";
+import { updatePage } from "../../components/pages/redux/pagesActions";
```
Comment (Saurabh186) [inline src/appPages/page/page.jsx:27]: aise kyu use karne pade nestjs me i think @krke bhi kar sakte hai

File: src/components/collectionVersions/collectionParentPages.jsx [hunk 2]
```diff
@@ -233,7 +233,12 @@ const CollectionParentPages = (props) => {
               <div className={`d-flex align-items-end ${isOnPublishedPage() ? 'w-100 cl-public-page' : 'cl-name'} `} onClick={(e) => handleParentPageClick(e, expanded)}>
                 <div className='d-flex td-name ml-1 align-items-center'>
                   <span className={`${isOnPublishedPage() ? 'versionChovron' : 'versionChovron icon-header'} d-flex justify-content-center`} onClick={(e) => handleToggle(e, props.rootParentId)}>
-                    <IconButtons variant='sm'><MdExpandMore size={13} className={`collection-icons-arrow d-none ${isOnPublishedPage() ? 'bg-white' : ''}`} /></IconButtons>
+                    <IconButtons variant='sm'>
+                      {isOnPublishedPage() ?
+                        (pages[pages[props.rootParentId]?.child?.length === 1 ? defaultVersionId : selectedVersionId]?.child?.length !== 0)
+                        && <MdExpandMore size={13} className={`collection-icons-arrow d-none ${isOnPublishedPage() ? 'bg-white' : ''}`} />
+                        : <MdExpandMore size={13} className={`collection-icons-arrow d-none ${isOnPublishedPage() ? 'bg-white' : ''}`} />}
```
Comment (Saurabh186) [inline src/components/collectionVersions/collectionParentPages.jsx:240]: yeh dono same hai aise kyu likhna pada 

<IconButtons variant="sm">
  {isOnPublishedPage() && (
    pages[pages[props.rootParentId]?.child?.length === 1 ? defaultVersionId : selectedVersionId]?.child?.length !== 0 && (
      <MdExpandMore 
        size={13} 
        className={`collection-icons-arrow d-none ${isOnPublishedPage() ? 'bg-white' : ''}`} 
      />
    )
  )}
</IconButtons>
isko use krke dekho ek bar test kar lena mai aise code likh diya uper ka dekh ke

File: src/components/collectionVersions/collectionParentPages.jsx [hunk 3]
```diff
@@ -233,7 +233,12 @@ const CollectionParentPages = (props) => {
               <div className={`d-flex align-items-end ${isOnPublishedPage() ? 'w-100 cl-public-page' : 'cl-name'} `} onClick={(e) => handleParentPageClick(e, expanded)}>
                 <div className='d-flex td-name ml-1 align-items-center'>
                   <span className={`${isOnPublishedPage() ? 'versionChovron' : 'versionChovron icon-header'} d-flex justify-content-center`} onClick={(e) => handleToggle(e, props.rootParentId)}>
-                    <IconButtons variant='sm'><MdExpandMore size={13} className={`collection-icons-arrow d-none ${isOnPublishedPage() ? 'bg-white' : ''}`} /></IconButtons>
+                    <IconButtons variant='sm'>
+                      {isOnPublishedPage() ?
```
Comment (Saurabh186) [inline src/components/collectionVersions/collectionParentPages.jsx:237]: yeh iconbutton ke baha ayega

File: src/components/endpoints/hostContainer.jsx [hunk 4]
```diff
@@ -154,9 +154,9 @@ class HostContainer extends Component {
       try {
         parsedData.data = JSON.parse(parsedData.data)
       } catch (e) { }
-      const contentType = (parsedData.headers?.['Content-Type'] || parsedData.headers?.['content-type'])?.toLowerCase()
+      let contentType = (parsedData.headers?.['Content-Type'] || parsedData.headers?.['content-type'])?.toLowerCase()
```
Comment (Saurabh186) [inline src/components/endpoints/hostContainer.jsx:157]: let kyu use krna pada

File: src/components/publicEndpoint/publicEndpoint.jsx [hunk 5]
```diff
@@ -421,7 +421,7 @@ class PublicEndpoint extends Component {
         : this.props.collections[collectionId]?.docProperties?.defaultLogoUrl
       const docTitle = this.props.collections[collectionId]?.docProperties?.defaultTitle
       // setTitle(docTitle)
-      setFavicon(docFaviconLink)
+      // setFavicon(docFaviconLink)
```
Comment (Saurabh186) [inline src/components/publicEndpoint/publicEndpoint.jsx:424]: tumhe lagta hai yeh code kaam nahi ana toh htado

File: src/components/tiptapEditor/tiptap.jsx [hunk 6]
```diff
@@ -480,106 +523,41 @@ export default function Tiptap({ provider, ydoc, isInlineEditor, disabled, initi
...
-              <FaRulerHorizontal className=' mr-4 ml-2' size={30} />
-              <div>
-                <span className="d-flex font-14 fw-500">Horizontal Rule</span>
-                <span className="menu-description mt-1 font-12">Visually divide blocks</span>
-              </div>
-            </li>
-          </ul>
+          <div className="slash-menu-container">
+            <div className="search-bar d-flex align-items-center p-16  m-2">
+              <span className='search-icon mr-2 ml-2 align-items-center justify-content-between'><BsSearch size={16} /></span>
```
Comment (Saurabh186) [inline src/components/tiptapEditor/tiptap.jsx:535]: isme display flex kaha hai jo align-items-center justify-content-between

File: src/components/tiptapEditor/tiptap.jsx [hunk 7]
```diff
@@ -480,106 +523,41 @@ export default function Tiptap({ provider, ydoc, isInlineEditor, disabled, initi
...
-            <li className='align-items-center d-flex  cursor-pointer px-2 py-2' tabIndex="0" ref={el => slashMenuRefs.current[12] = el} onClick={() => insertBlock('rule')} >
-              <FaRulerHorizontal className=' mr-4 ml-2' size={30} />
-              <div>
-                <span className="d-flex font-14 fw-500">Horizontal Rule</span>
-                <span className="menu-description mt-1 font-12">Visually divide blocks</span>
-              </div>
-            </li>
-          </ul>
+          <div className="slash-menu-container">
+            <div className="search-bar d-flex align-items-center p-16  m-2">
```
Comment (Saurabh186) [inline src/components/tiptapEditor/tiptap.jsx:534]: p-16 kya hai???

File: src/store/store.js [hunk 8]
```diff
@@ -18,7 +18,8 @@ import organizationReducer from '@/components/auth/redux/organizationRedux/organ
 import automationReducer from '@/components/collections/runAutomation/redux/runAutomationReducer'
 import createNewPublicEnvReducer from '@/components/publishDocs/redux/publicEnvReducer'
 
-// const storeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
+// const composeEnhancers =
```
Comment (Saurabh186) [inline src/store/store.js:21]: comment kyu need nahi hai remove it

File: src/components/tiptapEditor/tiptap.scss [hunk 9]
```diff
@@ -28,24 +28,53 @@ $scrollbar-color: #f1f1f1;
 }
 
 .slash-menu {
-    width: 260px;
     border: 1px solid $border-color ;
     box-shadow: 0 4px 12px rgba(0, 0, 0, 0.1);
     border-radius: 4px;
     z-index: 1;
 
+    .search-icon {
```
Comment (Saurabh186) [inline src/components/tiptapEditor/tiptap.scss:36]: maine bola hai yaar scss me likho chatGPT css deta hai aur bahut se css use ke nahi aur inke class usse kro

Comment (Saurabh186) [reply src/components/tiptapEditor/tiptap.scss:36]: sab dekh lena niche jo jo use huye hai

**Review by Saurabh186 — CHANGES_REQUESTED** [review]

File: src/appPages/page/page.jsx (hunk 1 above)
Comment (ViaSocket-Git) [reply src/appPages/page/page.jsx:27]: are vo component hi change hai dekho to. maine thodi kiya hai automatically liya hai usne

File: src/components/collectionVersions/collectionParentPages.jsx (hunk 2 above)
Comment (ViaSocket-Git) [reply src/components/collectionVersions/collectionParentPages.jsx:240]: conditiion dekho thik se different hai vo

File: src/store/store.js (hunk 8 above)
Comment (goutam01mehta) [reply src/store/store.js:21]: we have to left it commented but why this file has changes ? jo comment tha wo wesa ka wesa hona chahie tha

File: src/components/collectionVersions/collectionParentPages.jsx (hunk 3 above)
Comment (ViaSocket-Git) [reply src/components/collectionVersions/collectionParentPages.jsx:237]: ??

File: src/components/endpoints/hostContainer.jsx (hunk 4 above)
Comment (ViaSocket-Git) [reply src/components/endpoints/hostContainer.jsx:157]: mera nahi hai ye change

File: src/components/publicEndpoint/publicEndpoint.jsx (hunk 5 above)
Comment (ViaSocket-Git) [reply src/components/publicEndpoint/publicEndpoint.jsx:424]: okay

File: src/components/collectionVersions/collectionParentPages.jsx (hunk 3 above)
Comment (Saurabh186) [reply src/components/collectionVersions/collectionParentPages.jsx:237]: Spoon feeding thodi krunga

**Review by Saurabh186 — APPROVED** [review]

**Pull request closed without merging**